packages/test-runner-intercept/src/interceptModulePlugin.ts

```diff
@@ -0,0 +1,84 @@
+import { Plugin } from '@web/dev-server-core';
+import { readFile } from 'fs/promises';
+import { parse } from 'es-module-lexer';
+import { createResolveImport, ResolveImport } from './createResolveImport';
+import { stripColor } from './stripColor';
+
+/**
+ * Plugin that allows the interception of modules
+ */
+export function interceptModulePlugin(): Plugin {
+  const paths: string[] = [];
+
+  let resolveImport: ResolveImport;
+  return {
+    name: 'intercept-module',
+
+    serverStart(params) {
+      resolveImport = createResolveImport(params, this);
+    },
+    async serve(context) {
+      if (context.path.startsWith('/__intercept__/')) {
+        let body;
+        try {
+          const source = context.path.replace('/__intercept__/', '');
```

Comment:
how is this impacting tests running in separate tabs / concurrent?

PS: despite some issues with latest Chrome, we might still need to support that

Reply:
This path can be retrieved by multiple tabs. The generated code is concurrency proof. The result of the generation will only base itself on the original code of the module.
Any tab specific / test specific change is done in the browser context of a single tab. The changed code for an intercepted  module is in memory.

Reply:
let me clarify how it works and correct me where I'm wrong

- your test in Tab 1 requests intercepted module which makes a request to backend (WDS) which caches intercepted module in memor
- your test in Tab 2 fetches same module, although didn't request the intercepted one, but gets intercepted, because WDS had this cached in memory

isn't that an issue?

Reply:
Ah yes, in that sense.

Indeed, tab 2 will get the intercepted module. It should not pose a problem though. Tab 2 receives the interception, but the generated code of the intercepted module doesn't alter anything by default. The rewiring is done at runtime within the browser

Reply:
> The rewiring is done at runtime within the browser

right, so Test 2 should not notice any change in the module
lgmt!